test(textlint-formatter): `mocha` runs fine while `tsc` cannot compile the script

**What version of textlint are you using?**

Current tip of the master (0bb11b6778f49ecdb07292c2c564a4b76b7aa8b5)

**What did you do? Please include the actual source code causing the issue.**

I just found that my WebStorm shows red sign on `textlint-formatter-test.ts` even though I just finished all tests (`yarn run test:all`) on fresh docker image with success.

**What did you expect to happen?**

`textlint-formatter-test.ts` should fail.

**What actually happened? Please include the actual, raw output from textlint.**

After quick look here you can see simple scenario—`mocha` runs fine while `tsc` cannot compile the script.

```sh
$ cd packages/textlint-formatter
$ $(npm bin)/tsc -p test
test/textlint-formatter-test.ts(21,42): error TS2345: Argument of type '{ filePath: string; messages: { type: string; ruleId: string; line: number; column: number; index...' is not assignable to parameter of type 'TextlintResult[]'.
  Type '{ filePath: string; messages: { type: string; ruleId: string; line: number; column: number; index...' is not assignable to type 'TextlintResult'.
    Types of property 'messages' are incompatible.
      Type '{ type: string; ruleId: string; line: number; column: number; index: number; message: string; }[]' is not assignable to type 'TextlintMessage[]'.
        Type '{ type: string; ruleId: string; line: number; column: number; index: number; message: string; }' is not assignable to type 'TextlintMessage'.
          Property 'severity' is missing in type '{ type: string; ruleId: string; line: number; column: number; index: number; message: string; }'.
test/textlint-formatter-test.ts(55,42): error TS2345: Argument of type '({ filePath: string; messages: { type: string; ruleId: string; line: number; column: number; inde...' is not assignable to parameter of type 'TextlintResult[]'.
  Type '{ filePath: string; messages: { type: string; ruleId: string; line: number; column: number; index...' is not assignable to type 'TextlintResult'.
    Type '{ filePath: string; messages: { type: string; ruleId: string; line: number; column: number; index...' is not assignable to type 'TextlintResult'.
      Types of property 'messages' are incompatible.
        Type '{ type: string; ruleId: string; line: number; column: number; index: number; message: string; }[]' is not assignable to type 'TextlintMessage[]'.
          Type '{ type: string; ruleId: string; line: number; column: number; index: number; message: string; }' is not assignable to type 'TextlintMessage'.
            Property 'severity' is missing in type '{ type: string; ruleId: string; line: number; column: number; index: number; message: string; }'.
```


```sh
$ $(npm bin)/mocha test/textlint-formatter-test.ts


  textlint-formatter-test
    createFormatter
      ✓ should return formatter function (51ms)
      ✓ run all formatter (217ms)
      formatter
        ✓ should return output text
    getFormatterList
      ✓ should return list of formatter(s)


  4 passing (291ms)
```

Added `cache: false` to [`test/ts-setup.js`](https://github.com/textlint/textlint/blob/0bb11b6778f49ecdb07292c2c564a4b76b7aa8b5/packages/textlint-formatter/test/ts-setup.js) following [ts-node Configuration Options](https://github.com/TypeStrong/ts-node#configuration-options) but no clue.

Commit 139c7cbaf85506a4c7771b32fd9f4d736910857f changed `TextlintMessage` interface and made `severity` property [not optional](https://github.com/textlint/textlint/commit/139c7cbaf85506a4c7771b32fd9f4d736910857f#diff-d686a97daacd3bd8e8b66d37dcde7ae3L190). Since [`textlint-formatter-test.ts`](https://github.com/textlint/textlint/blob/0bb11b6778f49ecdb07292c2c564a4b76b7aa8b5/packages/textlint-formatter/test/textlint-formatter-test.ts) does not specify `severity` it should fail. It might be due to my setup or steps to test, and also changing the test is easy, but I would like to know why this happens.

--- a/packages/textlint-formatter/test/textlint-formatter-test.ts
+++ b/packages/textlint-formatter/test/textlint-formatter-test.ts
@@ -28,7 +28,8 @@ describe("textlint-formatter-test", function() {
                                 line: 1,
                                 column: 23,
                                 index: 0,
-                                message: "Expected a semicolon."
+                                message: "Expected a semicolon.",
+                                severity: 2
                             }
                         ]
                     }
@@ -62,7 +63,8 @@ describe("textlint-formatter-test", function() {
                                 line: 1,
                                 column: 1,
                                 index: 0,
-                                message: "0 pattern."
+                                message: "0 pattern.",
+                                severity: 2
                             },
                             {
                                 type: "lint",
@@ -70,7 +72,8 @@ describe("textlint-formatter-test", function() {
                                 line: 2,
                                 column: 26,
                                 index: 0,
-                                message: "Expected a semicolon."
+                                message: "Expected a semicolon.",
+                                severity: 2
                             },
                             {
                                 type: "lint",
@@ -78,7 +81,8 @@ describe("textlint-formatter-test", function() {
                                 line: 1,
                                 column: 21,
                                 index: 0,
-                                message: "Expected a semicolon."
+                                message: "Expected a semicolon.",
+                                severity: 2
                             },
                             {
                                 type: "lint",
@@ -86,7 +90,8 @@ describe("textlint-formatter-test", function() {
                                 line: 2,
                                 column: 26,
                                 index: 0,
-                                message: "Expected a semicolon."
+                                message: "Expected a semicolon.",
+                                severity: 2
                             }
                         ]
                     },
@@ -103,8 +108,7 @@ describe("textlint-formatter-test", function() {
                                 ruleId: "foo",
                                 fix: {
                                     range: [40, 45],
-                                    text: "fixed 1",
-                                    isAbsolute: false
+                                    text: "fixed 1"
                                 }
                             }
                         ]

--- a/packages/textlint-formatter/test/ts-setup.js
+++ b/packages/textlint-formatter/test/ts-setup.js
@@ -1,3 +1,4 @@
 require("ts-node").register({
-    project: __dirname
+    project: __dirname,
+    typeCheck: true
 });

